python/tank/bootstrap/manager.py

```diff
@@ -1118,6 +1124,60 @@ def _report_progress(self, progress_callback, progress_value, message):
             # Call the old style progress callback with signature (message, current_index, maximum_index).
             progress_callback(message, None, None)
 
+    def _process_bundle_cache_purge(self, progress_callback):
+        """
+        Check for unused bundles in bundle cache and delete them.
+
+        :param progress_callback: Callback function that reports back on the engine startup progress.
+        """
+
+        if os.environ.get('TK_BUNDLE_USAGE_TRACKING_NO_DELETE', None):
+            log.info("TK_BUNDLE_USAGE_TRACKING_DISABLE true, bundle usage tracking disabled.")
+        else:
+            try:
+                log.info("Checking bundle cache for unused bundles...")
+                # TODO: make global constant
+                days_since_last_usage = 30
+
+                bundle_path_list = bundle_cache_usage_mgr.get_unused_bundles(days_since_last_usage)
+                bundle_count = len(bundle_path_list)
+                purge_counter = 1
+                for bundle_path in bundle_path_list:
+                    version_str = os.path.basename(bundle_path[1])
+                    module_name = os.path.basename(os.path.dirname(bundle_path[1]))
+
+                    if os.environ.get('TK_BUNDLE_USAGE_TRACKING_NO_DELETE', None):
+                        message = "Warning '%s'version %s which was not used in last %d day%s (%d of %d)." % (
+                            module_name,
+                            version_str,
+                            int(days_since_last_usage),
+                            "s" if int(days_since_last_usage) > 1 else "",
+                            purge_counter,
+                            bundle_count
+                        )
+
+                    else:
+                        message = "Purging '%s'version %s which was not used in last %d day%s (%d of %d)." % (
+                            module_name,
+                            version_str,
+                            int(days_since_last_usage),
+                            "s" if int(days_since_last_usage) > 1 else "",
+                            purge_counter,
+                            bundle_count
+                        )
+                        #TODO: re-enable below
+                        bundle_cache_usage.purge_bundle(bundle_path)
+
+                    log.info(message)
+                    progress_value = float(purge_counter) / float(bundle_count)
+                    self._report_progress(progress_callback, progress_value, message)
+
+                    purge_counter += 1
+
+            except Exception as e:
```

Comment:
these kinds of blanket-catch-all are not good python practice. Instead, make sure that for example `bundle_cache_usage.purge_bundle(bundle_path)` logs a warning rather than raising an exception in case there is an error when trying to delete something.